pkg/spoa/root.go

```
@@ -400,12 +477,33 @@ func (s *Spoa) handleIPRequest(req *request.Request, mes *message.Message) {
 
 	ipStr := ipType.String()
 
-	r, _ = s.workerClient.GetIP(ipStr)
+	r, err = s.workerClient.GetIP(ipStr)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"ip":    ipStr,
+			"error": err,
+		}).Error("Failed to get IP remediation")
+		r = remediation.Allow // Safe default
+	}
 
 	if r < remediation.Unknown {
-		iso, _ := s.workerClient.GetGeoIso(ipStr)
-		if iso != "" {
-			r, _ = s.workerClient.GetCN(iso, ipStr)
+		iso, err := s.workerClient.GetGeoIso(ipStr)
+		if err != nil {
+			log.WithFields(log.Fields{
```

This may be verbose, so we might want to just silently ignore it